tests/integration/source_code/test_queries.py

```diff
@@ -1,60 +1,77 @@
-from databricks.labs.ucx.hive_metastore.table_migration_status import TableMigrationIndex
-from databricks.labs.ucx.source_code.directfs_access import DirectFsAccessCrawler
-from databricks.labs.ucx.source_code.queries import QueryLinter
-from databricks.labs.ucx.source_code.used_table import UsedTablesCrawler
+from databricks.labs.lsql.backends import Row
 
+from databricks.labs.ucx.source_code.base import DirectFsAccess, LineageAtom, UsedTable
 
-def test_query_linter_lints_queries_and_stores_dfsas_and_tables(
-    simple_ctx, ws, sql_backend, make_query, make_dashboard
-):
-    queries = [make_query(sql_query="SELECT * from csv.`dbfs://some_folder/some_file.csv`")]
-    dashboards = [make_dashboard(query=queries[0])]
-    queries.append(make_query(sql_query="SELECT * from some_schema.some_table"))
-    dashboards.append(make_dashboard(query=queries[1]))
-    linter = QueryLinter(
-        ws,
-        sql_backend,
-        simple_ctx.inventory_database,
-        TableMigrationIndex([]),
-        simple_ctx.directfs_access_crawler_for_queries,
-        simple_ctx.used_tables_crawler_for_queries,
-        None,
+
+def test_query_linter_lints_queries_and_stores_dfsas_and_tables(simple_ctx) -> None:
+    query_with_dfsa = simple_ctx.make_query(sql_query="SELECT * from csv.`dbfs://some_folder/some_file.csv`")
+    dashboard_with_dfsa = simple_ctx.make_dashboard(query=query_with_dfsa)
+    # Lakeview dashboard expects a string, not a legacy query
+    dashboard_with_used_table = simple_ctx.make_lakeview_dashboard(query="SELECT * FROM some_schema.some_table")
+
+    simple_ctx.query_linter.refresh_report()
+
+    problems = list(simple_ctx.sql_backend.fetch("SELECT * FROM query_problems", schema=simple_ctx.inventory_database))
+    assert problems == [
+        Row(
+            dashboard_id=dashboard_with_dfsa.id,
+            dashboard_parent=dashboard_with_dfsa.parent,
+            dashboard_name=dashboard_with_dfsa.name,
+            query_id=query_with_dfsa.id,
+            query_parent=query_with_dfsa.parent,
+            query_name=query_with_dfsa.name,
+            code='direct-filesystem-access-in-sql-query',
+            message='The use of direct filesystem references is deprecated: dbfs://some_folder/some_file.csv',
+        )
+    ]
+
+    dfsas = list(simple_ctx.directfs_access_crawler_for_queries.snapshot())
+    # By comparing the element instead of the list the `field(compare=False)` of the dataclass attributes take effect
+    assert len(dfsas) == 1, "Expected one DFSA"
+    assert dfsas[0] == DirectFsAccess(
```

Comment:
Same suggestion as elsewhere, although to be clear it's not a strong suggestion but rather something to consider.